safep/TI.py

```diff
@@ -49,8 +65,53 @@ def plot_TI(cumulative, perWindow, width=8, height=4, PDFtype='KDE', hystLim=(-1
     
     return fig, [cumAx,eachAx] 
 
-def make_harmonicWall(FC=10, targetFC=0, targetFE=1, upperWalls=1, schedule=None, numSteps=1000, targetEQ=500, name='HW', lowerWalls=None):
-    HW = {'name':name, 'targetFC':targetFC, 'targetFE':targetFE, 'FC':FC, 'upperWalls':upperWalls, 'schedule':schedule, 'numSteps':numSteps, 'targetEQ':targetEQ, 'lowerWalls':lowerWalls}
+def make_harmonicWall(FC=10, targetFC=0, targetFE=1, upperWalls=1, schedule=None, numSteps=1000, targetEQ=500, name='HW', lowerWalls=None, lambdaExp=1., decoupling=None):
+    '''
+    This method should rarely be used, as long as a log file exists that can be parsed by parse_Colvars_log
+    and the result fed to make_harmonicWall_from_Colvars
+    '''
+
+    # Heuristic to set the default value of the decoupling parameter
+    if decoupling is None:
+        decoupling = (targetFC == 0 and FC != 0)
+
+    HW = {'name':name, 'targetForceConstant':targetFC, 'lambdaExponent':targetFE, 'forceConstant':FC, 'upperWalls':upperWalls, 'schedule':schedule,
+           'targetNumSteps':numSteps, 'targetEquilSteps':targetEQ, 'lowerWalls':lowerWalls, 'lambdaExp':lambdaExp, 'decoupling':decoupling}
+
+    return HW
+
+def make_harmonicWall_from_Colvars(w):
+    '''
+    Input: a restraint configuration as a dict produced by parse_Colvars_log
+    '''
+
+    if (w['key'] != 'harmonicwalls'):
+        k = w['key']
+        print(f'Error: bias is not a harmonic wall (key: {k})')
+        return None
+
+    CVs = w['colvars'].strip("{}").replace(",", "").split()
+    if len(CVs) != 1:
+        print(f'Error: bias does not act on exactly one colvar (cvs: {CVs})')
+        return None

```

Comment:
I think this should be a `raise RuntimeError (f'Error ...')` or maybe `ValueError`
That will terminate execution of any script or cell of a jupyter notebook. Would that be a problem, or do we expect this function to return None sometimes?

Reply:
I think you're right, this should raise an exception.